jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ResourceHandler.java

```diff
@@ -99,20 +100,18 @@ else if (_baseResource.isAlias())
 
         setMimeTypes(context == null ? MimeTypes.DEFAULTS : context.getMimeTypes());
 
-        _byteBufferPool = getByteBufferPool(context);
+        ByteBufferPool.Sized byteBufferPool = new ByteBufferPool.Sized(findByteBufferPool(), _useDirectByteBuffers, _byteBufferSize);
         ResourceService resourceService = getResourceService();
-        resourceService.setHttpContentFactory(newHttpContentFactory());
+        resourceService.setHttpContentFactory(newHttpContentFactory(byteBufferPool));
         resourceService.setWelcomeFactory(setupWelcomeFactory());
         if (getStyleSheet() == null)
             setStyleSheet(getServer().getDefaultStyleSheet());
 
         super.doStart();
     }
 
-    private ByteBufferPool getByteBufferPool(Context context)
+    private ByteBufferPool findByteBufferPool()
```

Comment:
Feels weird to have to find the BBP, rather than just passing it in at construction time.

I'd prefer to have a `BBP.Sized` as a parameter to the constructor of `ResourceHandler`, as it is a necessary component.

This would have the benefit of removing the size/direct fields, as they are encapsulated in `BBP.Sized`.

Reply:
The way the ServerConnector handles this is to have a full constructor taking a buffer pool, plus one without that passes null.   If null is passed, then then the server is looked at for its buffer pool.

We should have the same logic here.

Reply:
I'm not against adding a constructor that expects a `ByteBufferPool.Sized`, but that would clash with the two properties I added (`_byteBufferSize` and `_useDirectByteBuffers`) that specify the size of the buffers when the pool is taken from the server. There would be some confusion between the size/directness of the properties and the ones of the sized pool.

Or we could add a constructor that takes a non-sized pool, but then the API makes you control the pool both via a constructor argument _and_ a couple properties; that's not very elegant and begs the question _why not passing pool + size + directness as ctor args_ which leads to _why not requiring a sized pool_ which sends us back to square 1.

Reply:
I would remove those two properties.  Just have an constructor that that takes a Sized pool.   If that is null then you look for a buffer pool in the server and you can default the buffer size and directness (or look for the first connector and take them from that).

Reply:
Sounds alright, let me do that.